### DISCUSSION TITLE
Add rule to enforce string quoting style

### DISCUSSION THREAD

**Discussion by nre-ableton** [ansible/ansible-lint#3937] [New rules]
Although [Ansible's YAML documentation](https://docs.ansible.com/ansible/latest/reference_appendices/YAMLSyntax.html) does not make specific recommendations on quoting style, it implicitly prefers unquoted strings except for the numerous YAML "gotchas". However, an info box in the [Windows usage documentation](https://docs.ansible.com/ansible/latest/os_guide/windows_usage.html#yaml-style) says:

> You should only quote strings when it is absolutely necessary or required by YAML, and then use single quotes.

IMHO this isn't a very good best-practice for playbooks, but I wish that whatever the preferred quoting style is, `ansible-lint` had some built-in capabilities to enforce it.

**Comment by nre-ableton** [comment]
(Somewhat related to https://github.com/ansible/ansible-lint/discussions/584)

**Comment by bbaassssiiee** [comment]
For more views: https://stackoverflow.com/questions/19109912/yaml-do-i-need-quotes-for-strings-in-yaml
